cns/service/main.go

```diff
@@ -869,6 +876,33 @@ func main() {
 		}
 	}
 
+	// Conditionally initialize and start the gRPC server
+	if cnsconfig.GRPCSettings.EnableGRPC {
+		// Define gRPC server settings
+		settings := grpc.GrpcServerSettings{
+			IPAddress: cnsconfig.GRPCSettings.GRPCServerIPAddress,
+			Port:      cnsconfig.GRPCSettings.GRPCServerPort,
+		}
+
+		// Initialize CNS service
+		cnsService := &grpc.CNSService{Logger: z}
+
+		// Create a new gRPC server
+		server, err := grpc.NewServer(settings, cnsService, z)
+		if err != nil {
+			logger.Errorf("[Listener] Could not initialize gRPC server: %v", err)
+			os.Exit(1)
+		}
+
+		// Start the gRPC server
+		go func() {
+			if err := server.Start(); err != nil {
+				logger.Errorf("[Listener] Could not start gRPC server: %v", err)
+				os.Exit(1)
```

Comment:
In general, you should try to use `os.Exit` as sparingly as possible. The process halts immedietly--it doesn't run `defer`s, so stuff that is intended to be cleaned up might not be. I would just `return` in both instances where you've used it.

Reply:
Good note! I have changed the` os.Exit ` to `return` in both instances thank you!